File: PARCIAL/Pregunta3/TransitSoft/SoftVenPersistance/DAOImpl/ReporteInfraccionDAOImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading;
using System.Threading.Tasks;
using SoftVen.DAOImpl;
using TransitSoftDomain;
using MySql.Data.MySqlClient;

namespace SoftVenPersistance.DAOImpl
{
    internal class ReporteInfraccionDAOImpl : DAOImplBase
    {
        private ReporteInfraccion reporteInfraccion;
        public ReporteInfraccionDAOImpl() : base("EX1_REPORTES_INFRACCIONES")
        {
            this.retornarLlavePrimaria = true;
            this.reporteInfraccion = null;
        }

        protected override void ConfigurarListaDeColumnas()
        {
            this.listaColumnas.Clear();
        }

        public int insertarReporteInfraccion(int conductorId, string paterno, string materno, string nombres, int vehiculoId, string placa, string marca, string modelo, int anho, int infraccionId, string descripcion, double monto, string gravedad)
        {
            reporteInfraccion = new ReporteInfraccion(conductorId, paterno, materno, nombres, vehiculoId, placa, marca, modelo, anho, infraccionId, descripcion, monto, gravedad);

            object reporteId = this.EjecutarProcedimientoConParametroOut(
                "INSERTAR_REGISTRO_INFRACCION",
                cmd => {
                    cmd.CommandType = System.Data.CommandType.StoredProcedure;

                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_CONDUCTOR_ID", conductorId));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_PATERNO"     , paterno));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_MATERNO"     , materno));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_NOMBRES"     , nombres));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_VEHICULO_ID" , vehiculoId));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_PLACA"       , placa));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_MARCA"       , marca));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_MODELO"      , modelo));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_ANHO"        , anho));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_INFRACCION_ID", infraccionId));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_DESCRIPCION" , descripcion));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_MONTO"       , monto));
                    cmd.Parameters.Add(new MySql.Data.MySqlClient.MySqlParameter("_GRAVEDAD"    , gravedad));
                },
                System.Data.DbType.Int32,
                "_REPORTE_ID"
            );
            return (int) reporteId;
        }
    }
}

File: Pregunta02/TransitSoft/TransitSoftAccesoDatos/src/main/java/pe/com/transitsoft/daoimpl/ReporteInfraccionDAOImpl.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.com.transitsoft.daoimpl;

import java.util.ArrayList;
import pe.com.transitsoft.dao.ReporteInfraccionDAO;
import pe.com.transitsoft.daoimpl.util.Columna;
import pe.com.transitsoft.daoimpl.util.Tipo_Dato;
import pe.com.transitsoft.modelo.ReporteInfraccion;

/**
 *
 * @author Yessica
 */
public class ReporteInfraccionDAOImpl extends DAOImplBase implements ReporteInfraccionDAO {
    private ReporteInfraccion reporteInfraccion;

    public ReporteInfraccionDAOImpl() {
        super("EX1_REPORTES_INFRACCIONES");
        this.retornarLlavePrimaria = true;
        this.reporteInfraccion = null;
    }

    @Override
    protected void configurarListaDeColumnas() {
        this.listaColumnas.add(new Columna("REPORTE_ID",      Tipo_Dato.ENTERO, true,  true));
        this.listaColumnas.add(new Columna("CONDUCTOR_ID",    Tipo_Dato.ENTERO, false, false));
        this.listaColumnas.add(new Columna("PATERNO",         Tipo_Dato.CADENA,  false, false));
        this.listaColumnas.add(new Columna("MATERNO",         Tipo_Dato.CADENA,  false, false));
        this.listaColumnas.add(new Columna("NOMBRES",         Tipo_Dato.CADENA,  false, false));
        this.listaColumnas.add(new Columna("VEHICULO_ID",     Tipo_Dato.ENTERO, false, false));
        this.listaColumnas.add(new Columna("PLACA",           Tipo_Dato.CADENA,  false, false));
        this.listaColumnas.add(new Columna("MARCA",           Tipo_Dato.CADENA,  false, false));
        this.listaColumnas.add(new Columna("MODELO",          Tipo_Dato.CADENA,  false, false));
        this.listaColumnas.add(new Columna("ANHO",            Tipo_Dato.ENTERO, false, false));
        this.listaColumnas.add(new Columna("INFRACCION_ID",   Tipo_Dato.ENTERO, false, false));
        this.listaColumnas.add(new Columna("DESCRIPCION",     Tipo_Dato.CADENA,  false, false));
        this.listaColumnas.add(new Columna("MONTO",           Tipo_Dato.REAL, false, false));
        this.listaColumnas.add(new Columna("GRAVEDAD",        Tipo_Dato.CADENA,  false, false));
    }

    @Override
    public Integer insertar(ReporteInfraccion dto) {
        this.reporteInfraccion = dto;
        return super.insertar();
    }

    @Override
    public Integer modificar(ReporteInfraccion dto) {
        this.reporteInfraccion = dto;
        return super.modificar();
    }

    @Override
    public Integer eliminar(ReporteInfraccion dto) {
        this.reporteInfraccion = dto;
        return super.eliminar();
    }

    @Override
    public ReporteInfraccion obtenerPorId(Integer id) {
        this.reporteInfraccion = new ReporteInfraccion();
        this.reporteInfraccion.setReporteId(id);
        super.obtenerPorId();
        return this.reporteInfraccion;
    }

    @Override
    public ArrayList<ReporteInfraccion> listarTodos() {
        return (ArrayList<ReporteInfraccion>) super.listarTodos();
    }
}